src/SelfService/Application/CapabilityApplicationService.cs

```diff
@@ -152,26 +158,52 @@ public async Task RegisterKafkaClusterAccessGranted(CapabilityId capabilityId, K
     }
 
     [TransactionalBoundary, Outboxed]
-    public async Task RequestCapabilityDeletion(CapabilityId capabilityId)
+    public async Task RequestCapabilityDeletion(CapabilityId capabilityId, UserId userId)
     {
         var capability = await _capabilityRepository.FindBy(capabilityId);
         if (capability is null)
         {
             throw EntityNotFoundException<Capability>.UsingId(capabilityId);
         }
         var modificationTime = _systemTime.Now;
-        capability.RequestDeletion();
+        capability.RequestDeletion(userId);
     }
 
     [TransactionalBoundary, Outboxed]
-    public async Task CancelCapabilityDeletionRequest(CapabilityId capabilityId)
+    public async Task CancelCapabilityDeletionRequest(CapabilityId capabilityId, UserId userId)
     {
         var capability = await _capabilityRepository.FindBy(capabilityId);
         if (capability is null)
         {
             throw EntityNotFoundException<Capability>.UsingId(capabilityId);
         }
         var modificationTime = _systemTime.Now;
-        capability.CancelDeletionRequest();
+        capability.CancelDeletionRequest(userId);
+    }
+
+    [TransactionalBoundary, Outboxed]
+    public async Task CheckPendingCapabilityDeletions()
+    {
+        var pendingCapabilities = await _capabilityRepository.GetAllPendingDeletionFor(days: PendingDaysUntilDeletion);
+        foreach (var capability in pendingCapabilities)
+        {
+            var message =
+                "*Capability Deletion Request*\n"
+                + "\nThe following capability has been pending deletion for 7 days and will be deleted in 24 hours:\n"
+                + $"Capability Name=\"{capability.Name}\" \\\n"
+                + $"Capability Id=\"{capability.Id}\" \\\n"
+                + $"Deletion Requested by user=\"{capability.ModifiedBy}\" \\\n"
+                + $"Originally Created by user=\"{capability.CreatedBy}\"";
+            var headers = new Dictionary<string, string>();
+            headers["CAPABILITY_NAME"] = capability.Name;
+            headers["CAPABILITY_ID"] = capability.Id;
+            headers["CAPABILITY_CREATED_BY"] = capability.CreatedBy;
+            headers["DELETION_REQUESTED_AT"] = capability.ModifiedAt.ToString();
+            headers["DELETION_REQUESTED_BY"] = capability.ModifiedBy;
+
+            await _ticketingSystem.CreateTicket(message, headers);
+
```

Comment:
Can you add logging here, it would be nice to see in the logs that the service is running and working.